custom/icds_reports/templates/icds_reports/icds_app/progress-report.directive.html

```diff
@@ -28,46 +28,21 @@ <h2>No Data</h2>
-                        <img style="width: 152px;" src="{% static 'img/maternal_and_child.png' %}"/>
-                        <div class="caption">Maternal and Child Nutrition</div>
-                    </div>
-                </div>
-                <div class="col-md-4">
-                    <div class="nav-block pointer col-sm-4 col-md-offset-2 col-md-8 fact-sheet-nav-interventions" ng-click="$ctrl.goToReport('interventions')">
-                        <img src="{% static 'img/interventions.png' %}"/>
-                        <div class="caption">Interventions</div>
-                    </div>
-                </div>
-                <div class="col-md-4">
-                    <div class="nav-block pointer col-sm-4 col-md-8" style="background-color: #ff8400;"  ng-click="$ctrl.goToReport('behavior_change')">
-                        <img style="width: 152px;" src="{% static 'img/behavior_change.png' %}"/>
-                        <div class="caption">Behavior Change</div>
+                <div class="col-md-4" ng-repeat="section in $ctrl.factSheetSections | limitTo: 3 ">
+                    <div class="nav-block pointer col-sm-4 col-md-8" ng-class="section.style" ng-click="$ctrl.goToReport(section.route)">
+                        <img style="width: 152px;" ng-src="{% static 'img/' %}{$section['image']$}"/>
+                        <div class="caption" ng-bind="section['name']"></div>
                     </div>
                 </div>
             </div>
             <div class="row" style="margin-top: 20px;">
-                <div class="col-md-4">
-                    <div class="nav-block pointer col-sm-4 col-md-offset-4 col-md-8 fact-sheet-nav-water" ng-click="$ctrl.goToReport('water_sanitation_and_hygiene')">
-                        <img src="{% static 'img/water_sanitation_hygiene.png' %}"/>
-                        <div class="caption">Water, Sanitation and Hygiene</div>
-                    </div>
-                </div>
-                <div class="col-md-4">
-                    <div class="nav-block pointer col-sm-4 col-md-offset-2 col-md-8" style="background-color: #00bdc5" ng-click="$ctrl.goToReport('demographics')">
-                        <img style="width: 100%" src="{% static 'img/demographics.png' %}"/>
-                        <div class="caption">Demographics</div>
-                    </div>
-                </div>
-                <div class="col-md-4">
-                    <div class="nav-block pointer col-sm-4 col-md-8" style="background-color: #9060c8;" ng-click="$ctrl.goToReport('all')">
-                        <img src="{% static 'img/all.png' %}"/>
-                        <div class="caption">All</div>
+                <div class="col-md-4" ng-repeat="section in $ctrl.factSheetSections | limitTo: (3 - $ctrl.factSheetSections.length)">
```

Comment:
this was a creative way to do this. didn't know that `limitTo` took negative numbers / how they behaved

Reply:
Yeah. I found it recently